src/app/components/form/form.component.ts

```diff
@@ -51,21 +52,37 @@ export class FormComponent implements OnInit, OnChanges, OnDestroy {
   private _showProgressBar: boolean;
   private langChangeSubscription: Subscription;
 
+  private domainName: string = '';
+  private routeParamsSubscription: Subscription;
+
   public history = {};
   public test = {};
   public disable_check_button = false;
   public form: FormGroup;
 
-  constructor(private formBuilder: FormBuilder, private translateService: TranslateService, private titleService: Title) {}
+  constructor(private activatedRoute: ActivatedRoute,
+    private formBuilder: FormBuilder,
+    private translateService: TranslateService,
+    private titleService: Title) {}
 
   ngOnInit() {
+    this.titleService.setTitle('Zonemaster');
+    this.generate_form();
+
+    this.routeParamsSubscription = this.activatedRoute.params.subscribe((params: Params) => {
+      if ( params['domain'] ) {
+        this.domainName = params['domain'];
+        this.form.controls.domain.setValue(this.domainName);
```

Comment:
Is the `domainName` property used anywhere else? If not would it make send to remove it?